### PR TITLE
fix(merge_issues): Fixes after merge - fixed AddUnitByChat

### PR DESCRIPTION
**Pull request by alexbojko**
### [ISSUE] 
After merging branches add unit by chat functionality was broken

### [SOLUTION]
Fixed AddUnitByChat - returned it to old flow (no chat JS init), added ChatNoJSInit view class to use it when we don't need JS init process.

### CODE REVIEW

File: mysite/chat/views.py [hunk 1]
```diff
@@ -357,6 +357,33 @@ def get(self, request, enroll_key, chat_id=None):
         )
 
 
+class ChatNoJSInit(object):
```
Comment (cmltaWt0) [inline mysite/chat/views.py:360]: Add docstring please.

File: mysite/chat/views.py [hunk 2]
```diff
@@ -357,6 +357,33 @@ def get(self, request, enroll_key, chat_id=None):
...
+    def get_or_init_chat(self, enroll_code, chat_id):
```
Comment (cmltaWt0) [inline mysite/chat/views.py]: Same as above.

File: mysite/chat/views.py [hunk 3]
```diff
@@ -357,6 +357,33 @@ def get(self, request, enroll_key, chat_id=None):
...
+        if chat_id:
+            chat = self.get_chat(
```
Comment (cmltaWt0) [inline mysite/chat/views.py]: This line w/o indentations will be more than 120 chars?

File: mysite/chat/views.py [hunk 4]
```diff
@@ -357,6 +357,33 @@ def get(self, request, enroll_key, chat_id=None):
...
+                **{'state__fsmNode__fsm__name': self.next_handler.FMS_name}
+            )
+        if not chat and enroll_code:
+            chat = self.create_new_chat(self.request, enroll_code, enroll_code.courseUnit)
+        if chat.message_set.count() == 0:
+            self.next_handler.start_point(
+                unit=enroll_code.courseUnit.unit, chat=chat, request=self.request
+            )
+        elif not chat.state:
+            next_point = None
```
Comment (cmltaWt0) [inline mysite/chat/views.py]: Do we need this extra variable?

**Review by cmltaWt0 — APPROVED** [review]

**Comment by rimikt** [comment]
Tests are failed.

**Review by cmltaWt0 — CHANGES_REQUESTED** [review]
Tests are failed.
Please look

**Pull request closed without merging**